rust/lance-encoding/benches/decoder.rs

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+use std::sync::Arc;
+
+use arrow_schema::{DataType, Field, TimeUnit};
+use criterion::{criterion_group, criterion_main, Criterion};
+use lance_encoding::encoder::encode_batch;
+use pprof::criterion::{Output, PProfProfiler};
+
+const PRIMITIVE_TYPES: &[DataType] = &[
+    DataType::Date32,
+    DataType::Date64,
+    DataType::Int8,
+    DataType::Int16,
+    DataType::Int32,
+    DataType::Int64,
+    DataType::UInt8,
+    DataType::UInt16,
+    DataType::UInt32,
+    DataType::UInt64,
+    DataType::Float16,
+    DataType::Float32,
+    DataType::Float64,
+    DataType::Decimal128(10, 10),
+    DataType::Decimal256(10, 10),
+    DataType::Timestamp(TimeUnit::Nanosecond, None),
+    DataType::Time32(TimeUnit::Second),
+    DataType::Time64(TimeUnit::Nanosecond),
+    DataType::Duration(TimeUnit::Second),
+    // The Interval type is supported by the reader but the writer works with Lance schema
+    // at the moment and Lance schema can't parse interval
+    // DataType::Interval(IntervalUnit::DayTime),
+];
+
+// Some types are supported by the encoder/decoder but Lance
+// schema doesn't yet parse them in the context of a fixed size list.
+const PRIMITIVE_TYPES_FOR_FSL: &[DataType] = &[
+    DataType::Int8,
+    DataType::Int16,
+    DataType::Int32,
+    DataType::Int64,
+    DataType::UInt8,
+    DataType::UInt16,
+    DataType::UInt32,
+    DataType::UInt64,
+    DataType::Float16,
+    DataType::Float32,
+    DataType::Float64,
+];
+
+fn bench_decode(c: &mut Criterion) {
+    let mut group = c.benchmark_group("decode_primitive");
+    for data_type in PRIMITIVE_TYPES {
+        let data = lance_datagen::gen()
+            .col(None, lance_datagen::array::rand_type(&DataType::Int32))
+            .into_batch_rows(lance_datagen::RowCount::from(1024 * 1024))
+            .unwrap();
+        let rt = tokio::runtime::Runtime::new().unwrap();
```

Comment:
nit: instead of intializing a runtime inside of a loop, could we move to a static one? Here's an example of using `std::sync::OnceLock` to accomplish this:

https://github.com/ion-elgreco/delta-rs/blob/4a0e47ee1f82cc65c78a97a5722890fb962f0e34/python/src/utils.rs#L8-L12

Reply:
Done.  Though I just moved the runtime creation to the top of the benchmark method (outside the loop) instead of making it static.